src/frontend/src/expr/pure.rs

```diff
@@ -187,7 +187,10 @@ impl ExprVisitor<bool> for ImpureAnalyzer {
                 x
             }
             // expression output is not deterministic
-            expr_node::Type::Vnode | expr_node::Type::Proctime | expr_node::Type::Udf => true,
+            expr_node::Type::Vnode
+            | expr_node::Type::Now
```

Comment:
... for example, this is actually a dead code path as there's no **function call** with type `Now`.